diff --git a/src/freedreno/vulkan/tu_device.cc b/src/freedreno/vulkan/tu_device.cc
index 8a7bcd1..f2d9a45 100644
--- a/src/freedreno/vulkan/tu_device.cc
+++ b/src/freedreno/vulkan/tu_device.cc
@@ -105,7 +105,7 @@ tu_physical_device_init(struct tu_physical_device *device)
-   physical_device->has_cached_coherent_memory = true;
+   physical_device->has_cached_coherent_memory = false;  // Dragon Falcon patch
